jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/TypeUtil.java

```diff
@@ -405,6 +405,27 @@ public static int parseInt(byte[] b, int offset, int length, int base)
         return value;
     }
 
+    /**
+     * Computes a new capacity for a collection willing to grow, based on its
+     * current capacity and its max capacity.
+     * @param currentCapacity the collection's current capacity
+     * @param maxCapacity the collection's max capacity
+     * @return the collection's new capacity, guaranteed to be &lt;= maxCapacity
+     */
+    public static int newCapacity(int currentCapacity, int maxCapacity)
```

Comment:
No, `TypeUtil` is about types, not capacities.
Move it to the only place it is used, i.e. BAQ.